### PR TITLE
Add the possibility to configure the order of comments

### PR DESCRIPTION
**Pull request by andywenk**
This patch introduces a new configuration parameter

  config.comment_order = 'created_at ASC'

The user will be able to configure if the comments
should be displayed ascending or descending. Furthermore
the user is able to choose the column which should be
used for ordering.

### CODE REVIEW

**Comment by andywenk** [comment]
sry - my failure - I have to admit that I was not able to get the test suite running and I hoped it will run. But this was a stupid mistake. Will update the PR in a second.

File: lib/active_admin/orm/active_record/comments/comment.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def self.find_for_resource_in_namespace(resource, namespace)
         resource_type: resource_type(resource),
         resource_id:   resource_id_cast(resource),
         namespace:     namespace.to_s
-      ).order('created_at ASC')
+      ).order(ActiveAdmin.application.comment_order)
```
Comment (timoschilling) [inline lib/active_admin/orm/active_record/comments/comment.rb]: This code need's to support different namespace config's.

File: lib/active_admin/application.rb [hunk 2]
```diff
@@ -105,6 +105,9 @@ def initialize
     # Set flash message keys that shouldn't show in ActiveAdmin
     inheritable_setting :flash_keys_to_except, ['timedout']
 
+    # Set the order and column to be used for ordering for comments
+    inheritable_setting :comment_order, "created_at ASC"
+
```
Comment (timoschilling) [inline lib/active_admin/application.rb]: This need's to be placed here: https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/orm/active_record/comments.rb

**Comment by andywenk** [comment]
Thanks for your comments @timoschilling . As this is my first PR for ActiveAdmin I would love to get a bit more explanation and just place this here and that there. A pointer to a similar example is good enough. Thank's a lot.

File: lib/active_admin/orm/active_record/comments/comment.rb (hunk 1 above)
Comment (timoschilling) [inline lib/active_admin/orm/active_record/comments/comment.rb]: This should work:
```ruby
ActiveAdmin.application.namespaces[namespace].comment_order
```

File: lib/active_admin/application.rb (hunk 2 above)
Comment (timoschilling) [inline lib/active_admin/application.rb]: All Comment's related settings are placed in the [comments.rb](https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/orm/active_record/comments.rb) file, simply place Line 109 in Line 10 of the comments.rb file.

Comment (timoschilling) [inline lib/active_admin/application.rb]: The setting should be plural named.

File: lib/active_admin/orm/active_record/comments.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@
 ActiveAdmin::Application.inheritable_setting :comments,                   true
 ActiveAdmin::Application.inheritable_setting :show_comments_in_menu,      true
 ActiveAdmin::Application.inheritable_setting :comments_registration_name, 'Comment'
+ActiveAdmin::Application.inheritable_setting :comments_order,             "created_at ASC"
```
Comment (andywenk) [inline lib/active_admin/orm/active_record/comments.rb]: will ignore this because it's the code style here ...

**Comment by andywenk** [comment]
Ok before I flood this more with not tested tests I have to get the local environment running. Please accept my apologise for this ...

File: lib/active_admin/orm/active_record/comments.rb (hunk 3 above)
Comment (timoschilling) [inline lib/active_admin/orm/active_record/comments.rb]: Thats ok

File: lib/active_admin/orm/active_record/comments/comment.rb [hunk 4]
```diff
@@ -29,7 +29,7 @@ def self.find_for_resource_in_namespace(resource, namespace)
         resource_type: resource_type(resource),
         resource_id:   resource_id_cast(resource),
         namespace:     namespace.to_s
-      ).order('created_at ASC')
+      ).order(ActiveAdmin.application.namespaces[namespace].comments_order)
```
Comment (timoschilling) [inline lib/active_admin/orm/active_record/comments/comment.rb]: Hint: namespace need's to be a Symbol

**Comment by timoschilling** [comment]
Don't worry about too many untested commits, you can squash them into one if you are done.

How can I help you with the test setup?

**Comment by andywenk** [comment]
thanks for your help @timoschilling. My setup ist rbenv and I installed ruby 2.1.1 (due to your hint, that I need Ruby < 2.2). RSpec is 3.1.7. When running `rake test` I get the error, that `ActiveAdmin.register(Category)` in `spec/rails_helper.rb` is an uninitialised constant. I am not sure if that still has sth. to do with wrong versions in my setup ...

	rake test
	/Users/andwen/.rbenv/versions/2.1.1/bin/ruby -I/Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib:/Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-support-3.1.2/lib 		
	/Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/exe/rspec --pattern spec/unit/\*\*/\*_spec.rb
			DEPRECATION WARNING: Support for Rails < 4.1.0 will be dropped. (called from warn at /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/2.1.0/forwardable.rb:183)
			Coverage report generated for RSpec to /Users/andwen/project/sumcumo/activeadmin/coverage. 1685 / 3324 LOC (50.69%) covered.
			/Users/andwen/project/sumcumo/activeadmin/spec/rails_helper.rb:9:in `load_defaults!': uninitialized constant ActiveAdminIntegrationSpecHelper::Category (NameError)
				from /Users/andwen/project/sumcumo/activeadmin/spec/rails_helper.rb:112:in `<top (required)>'
				from /Users/andwen/project/sumcumo/activeadmin/spec/unit/abstract_view_factory_spec.rb:1:in `require'
				from /Users/andwen/project/sumcumo/activeadmin/spec/unit/abstract_view_factory_spec.rb:1:in `<top (required)>'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `load'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `block in load_spec_files'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `each'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/configuration.rb:1105:in `load_spec_files'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:96:in `setup'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:84:in `run'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:69:in `run'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib/rspec/core/runner.rb:37:in `invoke'
				from /Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/exe/rspec:4:in `<main>'
			/Users/andwen/.rbenv/versions/2.1.1/bin/ruby -I/Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/lib:/Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-support-3.1.2/lib 		
			/Users/andwen/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rspec-core-3.1.7/exe/rspec --pattern spec/unit/\*\*/\*_spec.rb failed

**Comment by seanlinsley** [comment]
It looks like your test rails app is broken. Try running this: `rm -rf spec/rails`

**Comment by andywenk** [comment]
thanks @seanlinsley - I will give it a try as soon as I can find a free time slot ;-)

**Comment by andywenk** [comment]
just a quick update. I have hard times finding time to continue but plan to work on this next week. Sorry for the delay

**Comment by timoschilling** [comment]
any updates?

**Comment by andywenk** [comment]
sorry I could not find time but I still have it in my mind. If I get it working i will send a new PR

**Comment by timoschilling** [comment]
@andywenk you don't need that, I have finished this PR based on your code

**Comment by andywenk** [comment]
@timoschilling woah - awesome ;-) Thanks a lot!

**Pull request closed without merging**